ssh_proxy_server/forwarders/sftp.py

```diff
@@ -67,3 +69,43 @@ def write(self, offset, data):
         data = self.plugin.handle_data(data)
         self.writefile.write(data)
         return paramiko.SFTP_OK
+
+
+class SFTPHandlerReplacePlugin(SFTPHandlerPlugin):
+    """
+    Replaces a SFTP transmitted File during transit
+    """
+    @classmethod
+    def parser_arguments(cls):
+        cls.PARSER.add_argument(
+            '--sftp-replace',
+            dest='sftp_replacement_file',
+            required=True,
+            help='file that is used for replacement'
+        )
+
+    def __init__(self, sftp, filename):
+        super().__init__(sftp, filename)
+        logging.info("sftp file transfer detected: %s", filename)
+        logging.info("intercepting sftp file, replacement: %s", self.args.sftp_replacement_file)
+        self.file = None
+        self.replacement = open(self.args.sftp_replacement_file, "rb")
+        self.buf_size = None
+
+    def close(self):
+        self.replacement.close()
+
+    def handle_data(self, data):
```

Comment:
you have to pass the data length to the method:

```python
    def handle_data(self, data, length=None):
        return self.replacement.read(length)
```
To pass the legth parameter, you have to modify the 'handle_data' method from 'SFTPBaseHandle':

```python
class SFTPBaseHandle(paramiko.SFTPHandle):

    def read(self, offset, length):
        data = self.readfile.read(length)
        return self.plugin.handle_data(data, length)
```

To get the file replacement file size, you have to subclass  'ssh_proxy_server.interfaces' and edit the methods 'stat' and 'lstat':

```python
from paramiko.sftp_attr import SFTPAttributes

class ReplaceerSFTPProxyServerInterface(SFTPProxyServerInterface):

    def lstat(self, path):
        self.session.sftp_client_ready.wait()
        stat_remote = self.session.sftp_client.lstat(path)
        stat_replace = SFTPAttributes.from_stat(os.stat('/bin/bash'))
        stat_remote.st_size = stat_replace.st_size
        return stat_remote

    def stat(self, remotePath):
        self.session.sftp_client_ready.wait()
        stat_remote = self.session.sftp_client.stat(remotePath)
        stat_replace = SFTPAttributes.from_stat(os.stat('/bin/bash'))
        stat_remote.st_size = stat_replace.st_size
        return stat_remote
```

You should also add a new cmd argument to the main-Function in cli.py, where you can specify the ProxyServerInterface.

Reply:
http://docs.paramiko.org/en/stable/api/sftp.html